syscalls.go

```diff
@@ -185,6 +186,10 @@ func bpfProgAlter(cmd int, attr *bpfProgAlterAttr) error {
 
 func bpfMapCreate(attr *bpfMapCreateAttr) (*internal.FD, error) {
 	fd, err := internal.BPF(_MapCreate, unsafe.Pointer(attr), unsafe.Sizeof(*attr))
+	if xerrors.Is(err, os.ErrPermission) {
+		return nil, xerrors.Errorf("permission denied or insufficient rlimit to lock memory for map '%s'", attr.mapName)
```

Comment:
mapName may be truncated or empty. I think callers will already include the name if possible, so skip the format here?

Reply:
Ah, the call to `newMapWithBTF` already contains the format, got it.